code/game/objects/items/grenades/chem_grenade.dm

```diff
@@ -31,10 +31,14 @@
 	..()
 	if(user.can_see_reagents())
 		if(beakers.len)
-			to_chat(user, "<span class='notice'>You scan the grenade and detect the following reagents:</span>")
-			for(var/obj/item/reagent_containers/glass/G in beakers)
-				for(var/datum/reagent/R in G.reagents.reagent_list)
-					to_chat(user, "<span class='notice'>[R.volume] units of [R.name] in the [G.name].</span>")
+			to_chat(user, "<span class='notice'>You scan the grenade and detect the following objects:</span>")
+			for(var/obj/item/reagent_containers/G in beakers)
+				to_chat(user, "<span class='notice'>A [G.name]. It contains:.</span>")
+				if(G.reagents.reagent_list.len)			

```

Comment:
you don't need this check before a loop

Reply:
I need this check somewhere so line 41 can happen if false. I don't see why not here.